gslib/tests/test_hmac.py

```diff
@@ -357,3 +403,168 @@ def test_hmac_fails_for_xml(self):
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': fake_cloudsdk_dir,
+      }):
+        mock_log_handler = self.RunCommand(
+            'hmac',
+            args=['delete', '-p', project, access_id],
+            return_log_handler=True)
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(('Gcloud Storage Command: {} alpha storage'
+                       ' hmac delete --project {} {}').format(
+                           os.path.join('fake_dir', 'bin', 'gcloud'), project,
+                           access_id), info_lines)
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
+                     'GetHmacKey',
+                     autospec=True)
+  def test_shim_translates_get_commannd(self, mock_get_hmac_key):
+    fake_cloudsdk_dir = 'fake_dir'
+    project = 'test-project'
+    access_id = 'fake123456789'
+    with SetBotoConfigForTest([('GSUtil', 'use_gcloud_storage', 'True'),
+                               ('GSUtil', 'hidden_shim_mode', 'dry_run')]):
+      with SetEnvironmentForTest({
+          'CLOUDSDK_CORE_PASS_CREDENTIALS_TO_GSUTIL': 'True',
+          'CLOUDSDK_ROOT_DIR': fake_cloudsdk_dir,
+      }):
+        mock_log_handler = self.RunCommand(
+            'hmac',
+            args=['get', '-p', project, access_id],
+            return_log_handler=True)
+
+        info_lines = '\n'.join(mock_log_handler.messages['info'])
+        self.assertIn(('Gcloud Storage Command: {} alpha storage'
+                       ' hmac describe {} --project {} {}').format(
+                           os.path.join('fake_dir', 'bin', 'gcloud'),
+                           hmac._DESCRIBE_COMMAND_FORMAT, project, access_id),
+                      info_lines)
+
+  @mock.patch.object(cloud_api_delegator.CloudApiDelegator,
+                     'ListHmacKeys',
+                     autospec=True)
+  def test_shim_translates_hmac_list_command(self, mock_list_hmac_keys):
+    fake_cloudsdk_dir = 'fake_dir'
+    project = 'test-project'
+    service_account = (
+        'test.service.account@test_project.iam.gserviceaccount.com')
+    command_expected_list_command_format = (([
+        'list', '-a', '-u', service_account, '-p', project
+    ], ('Gcloud Storage Command: {} alpha storage'
+        ' hmac list {} --all --service-account {} --project {}').format(
+            os.path.join('fake_dir', 'bin', 'gcloud'),
+            hmac._LIST_COMMAND_SHORT_FORMAT, service_account, project)), ([
+                'list', '-a', '-u', service_account, '-l', '-p', project
+            ], ('Gcloud Storage Command: {} alpha storage'
+                ' hmac list {} --all --service-account {} --long --project {}'
+               ).format(os.path.join('fake_dir', 'bin',
+                                     'gcloud'), hmac._DESCRIBE_COMMAND_FORMAT,
+                        service_account, project)))
+    hmac_key = apitools_messages.HmacKeyMetadata(
```

Comment:
Why do you need a return value from the API for testing that a command translated?

Reply:
No need, I think mocking the RunCommand is a better option instead of mocking the API since the shim translation is being tested.

Reply:
Changed all unit tests for the shim translation using this suggestion, removed code that mocked the API.